crypto/evp/evp_enc.c

```diff
@@ -559,8 +559,7 @@ static int evp_cipher_init_skey_internal(EVP_CIPHER_CTX *ctx,
         }
     }
 
-    if (skey != NULL && skey->skeymgmt != NULL
-        && ctx->cipher->prov != skey->skeymgmt->prov) {
+    if (skey != NULL && ctx->cipher->prov != skey->skeymgmt->prov) {
```

Comment:
So skey->skeymgmt can never be NULL here?

Reply:
Yes, skey->skeymgmt can never be NULL. That is an invariant.